### ISSUE TITLE
⌨ Remove work email button name ambiguous

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#15172]
### ⌨ Accessibility and Expected Behaviour

The button to remove a work email should be more specific.

### 📋 Steps to Reproduce

1. Login as `applicant-employee@test.com`
2. Navigate to `/applicant/settings`
3. Open the accessibilty tree
4. Inspect the remove button for work email
5. Observe it is simply "Remove"

### 🕵️ Details

For AT users, this button is pretty ambiguous when used out of context (rotor menu when listing buttons).

It should have context added like "Remove work email"

### 📸 Screenshot

<img width="1234" height="347" alt="Image" src="https://github.com/user-attachments/assets/04211f30-1a03-444d-801a-6f726489aba8" />

### 🙋‍♀️ Proposed Solution

Update the text to be "Remove work email"

### ✅ Acceptance Criteria

- [ ] Work email removal trigger accessible name is not ambiguous

**Comment by gobyrne** [comment]
We have [well documented patterns](https://app.slack.com/client/T5ZFRSQ3V/unified-files/doc/F09CCL205JS) for how to handle this. The proposed solution is the best one. Let's go with that.
